### ISSUE TITLE
Use Spool instread of Enumerator when using twitter futures

### ISSUE THREAD

**Issue by benjumanji** [websudos/phantom#189]
Idea: Instead of going via the play lib when using twitter futures, use a `Spool`. Why? because if you are threading around context using twitter/util `Local` it will still be accessible inside of phantom. With this in place you could potentially reuse `Tracer` from finagle to annotate things inside of the query.

I would be very happy to engage in doing the work, I just would like to know if you be open to such a change. If there were also a bridge to turn a `Spool` into an `Enumerator` (which I think would be trivial) it wouldn't even necessarily mean breaking api changes, you'd just get some extra functionality.

**Comment by benjumanji** [comment]
Basically to get tracing, you could merely check if if finagle was on the class path, and if so, check for tracing enabled, and if so, add tracing data. How cool would that be?

**Comment by alexflav23** [comment]
@benjumanji There are several bumps in the road to achieve what you suggest, many thanks which we are already considering.

- Phantom does not have a dependency on Finagle for core functionality, but rather on Twitter util. As most users do not user Finagle we cannot increase dependencies that drastically, Finagle is not exactly a small project.

- We've been planning on completely breaking the two parts apart, e.g. the Scala API and the Finagle API by devising 2 modules. E.g if you really want to use finagle you need an extra dependency but somehow Twitter Futures are still first class citizens. Not sure what patterns is a best fit for this since you should ideally have the Twitter futures option enabled without any imports or any code changes.

- Benchmarking is extremely important, since everything winds up in the Enumerator. We would need to tightly control the bridge between ```Spool``` and ```Enumerator```, which I'm sure can be done with enough profiling.

Next there's the question of bandwidth, the usually less than stellar chapter. If you can get start working on this we are very happy to merge and the outcome will probably be a new ```phantom-finagle``` module that will pre-bundle tracing and other Finagle goodies. We could even leverage more of Finagle, like flags and the request filtering mechanism, should there be a use case for them.

Long story short, you have all the concerns listed.

**Comment by benjumanji** [comment]
Sounds cool. I would agree that the tracing stuff should / would live in its own module, so that only people that have finagle anyway on the class path should incur a cost. I have a few things to take care of, but I think end of next week I will start in earnest on it, because it would make my logging so much prettier to be able to pass things around with `com.twitter.finagle.context.Context`

**Comment by benjumanji** [comment]
Bewm: PR opened. I didn't want to go to sleep anyway,
